services/sort.service.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const CONFIG = require('../config/app.js');
+
+module.exports = SortService;
+
+function getTabByKeyVal(key, val, myArray, myarray2) {
+  if (key === undefined || val === undefined || (myArray === undefined && myarray2 === undefined) ) {
+    return undefined;
+  }
+  if (myArray !== undefined) {
+    for (var i = 0; i < myArray.length; i++) {
+      if (myArray[i][key] === val) {
+        return myArray[i];
+      }
+    }
+  }
+  if (myarray2 !== undefined) {
+    for (var i = 0; i < myarray2.length; i++) {
+      if (myarray2[i][key] === val) {
+        return myarray2[i];
+      }
+      else if (("aliases" in myarray2[i]))
+      {
+        for (var j = 0; j < myarray2[i]["aliases"].length; j++) {
+          if (myarray2[i]["aliases"][j] === val) {
+            return myarray2[i];
+          }
+        }
+      }
+    }
+  }
+  return undefined;
+}
+
+function getValIfValExistInArray(value, param) {
+  if (param["model"] === undefined && param["modelName"] === undefined)
+    return value;
+  var typeTab = getTabByKeyVal("name", value, param["model"], param["modelName"]);
+  if (typeTab !== undefined) {
+    return value;
+  }
+  else {
+    return (undefined);
+  }
+}
+
+function SortService(options) {
+  function checkSortParam(sorts, param)
+  {
+    var objRes = {};
+    if (typeof sorts === 'object')
```

Comment:
sorts is not an object, it is a string

eg: "?sort=created_at"
eg: "?sort=name"
eg: "?sort=-name" (inversed order)